relays/ethereum/src/ethereum_deploy_contract.rs

```diff
@@ -42,10 +43,12 @@ pub struct EthereumDeployContractParams {
 	pub sub_initial_authorities_set: Option<Vec<u8>>,
 	/// Initial header.
 	pub sub_initial_header: Option<Vec<u8>>,
+	/// Instance of the bridge pallet being synchronized.
+	pub instance: Box<dyn BridgeInstance>,
 }
 
-impl Default for EthereumDeployContractParams {
-	fn default() -> Self {
+impl EthereumDeployContractParams {
+	pub fn with_instance(instance: Box<dyn BridgeInstance>) -> Self {
```

Comment:
Docs?
```suggestion
	pub fn new(instance: Box<dyn BridgeInstance>) -> Self {
```
To be fair I'd prefer simply `new` instead of `with_instance`. I feel like `with_*` constructors/builder functions are useful if you want to parametrize some extra stuff that is set to default values in `new`. I.e. they allow customization over default `new` constructor. Here however we only have one way to create the params and I don't feel like `with_instance` adds any extra value. Looking at the std library we have:
https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.new
https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.with_capacity
but there is no `with_value` for things like `Box`, `Cell`, etc:
https://doc.rust-lang.org/std/sync/atomic/struct.AtomicUsize.html#method.new